src/dialogs/symlink.tsx

```diff
@@ -0,0 +1,254 @@
+                } catch (exc) {
+                    const err = exc as BasicError;
+                    setErrorMessage(err.toString());
+                    return;
+                }
+            } else {
+                cmd.push(target, symlinkName);
+            }
+        } else if (mode === "relative") {
+            cmd.push("--relative", target, symlinkName);
+        }
+
+        try {
+            await cockpit.spawn(cmd, {
+                err: "message",
+                directory: path,
+                ...owner && { superuser: "require" }
+            });
+            if (owner !== null) {
+                await cockpit.spawn(["chown", "--no-dereference", owner, symlinkName], {
+                    directory: path,
+                    superuser: "require"
+                });
+            }
+        } catch (exc) {
+            const err = exc as BasicError;
+            setErrorMessage(err.toString());
+            return;
+        }
+
+        dialogResult.resolve();
+    };
+
+    return (
+        <Modal
+          position="top"
+          title={fmt_to_fragments(_("Create link to $0"), <b>{selected.name}</b>)}
+          variant={ModalVariant.medium}
+          isOpen
+          className="file-symlink-modal"
+          onClose={() => dialogResult.resolve()}
+          footer={
+              <>
+                  <Button
+                    variant="primary"
+                    onClick={() => {
+                        const filename_valid = checkLinkName(symlinkName, cwdInfo?.entries || {}, selected);
+                        if (filename_valid) {
+                            setSymlinkError(filename_valid);
+                        } else {
+                            createLink();
+                        }
+                    }}
+                    isDisabled={errorMessage !== undefined ||
+                        symlinkError !== null ||
+                        cwdInfo === null ||
+                        user === null}
+                  >
+                      {_("Create link")}
+                  </Button>
+                  <Button variant="link" onClick={() => dialogResult.resolve()}>{_("Cancel")}</Button>
+              </>
+          }
+        >
+            {errorMessage &&
+            <InlineNotification
+              type="danger"
+              text={errorMessage}
+              isInline
+            />}
+            <Form
+              isHorizontal
+              onSubmit={e => {
+                  e.preventDefault();
+                  createLink();
+                  return false;
+              }}
+            >
+                <FormGroup fieldId="target-name" label={_("Target")}>
+                    <TextInput
+                      value={selected.name}
+                      readOnlyVariant="plain"
+                      id="target-name"
+                    />
+                </FormGroup>
+                <FormGroup fieldId="symlink-input" label={_("Symlink name")}>
```

Comment:
I find it a bit confusing that this is called `Symlink name` when it can be a full path rather than just name.
Not sure what would be a better alternative though. `Symlink path` doesn't sound all that good 🤔

Reply:
Maybe @garrett has an idea, naming comes from https://design.penpot.app/#/workspace?team-id=8ad824c0-6542-11eb-a7f8-eb0147e048f5&file-id=7e3dac9f-07b8-8055-8005-a7cbdf87f135&page-id=8cabb599-04b2-80d4-8005-06d4440bc451